usb/util.py

```diff
@@ -254,22 +302,15 @@ def get_string(dev, index, langid = None):
     The return value is the unicode string present in the descriptor.
     """
     from usb.control import get_descriptor
+    langids = dev.langids
+
+    if 0 == len(langids):
+        raise ValueError("get_string from a device that offers none")
```

Comment:
Here was my reasoning: USB spec explicitly requires that no get string descriptor request ever be sent to a device for a langid the device has not offered. If that is left for every app that "wants" to make sure, then every app had better "want" to make sure, or noncompliant USB traffic will be sent. If checked here, then pyusb avoids generating noncompliant traffic, and if any client code is making a buggy assumption or failing to check, it will be caught here.